homeassistant/components/habitica/config_flow.py

```diff
@@ -2,83 +2,167 @@
 
 from __future__ import annotations
 
+from http import HTTPStatus
 import logging
+from typing import Any
 
 from aiohttp import ClientResponseError
 from habitipy.aio import HabitipyAsync
 import voluptuous as vol
 
-from homeassistant.config_entries import ConfigFlow
-from homeassistant.const import CONF_API_KEY, CONF_NAME, CONF_URL
-from homeassistant.core import DOMAIN as HOMEASSISTANT_DOMAIN, HomeAssistant
-from homeassistant.exceptions import HomeAssistantError
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_PASSWORD,
+    CONF_URL,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import DOMAIN as HOMEASSISTANT_DOMAIN
 from homeassistant.helpers.aiohttp_client import async_get_clientsession
 from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
 
 from .const import CONF_API_USER, DEFAULT_URL, DOMAIN
 
-DATA_SCHEMA = vol.Schema(
+STEP_ADVANCED_DATA_SCHEMA = vol.Schema(
     {
         vol.Required(CONF_API_USER): str,
         vol.Required(CONF_API_KEY): str,
-        vol.Optional(CONF_NAME): str,
         vol.Optional(CONF_URL, default=DEFAULT_URL): str,
+        vol.Required(CONF_VERIFY_SSL, default=True): bool,
     }
 )
 
-_LOGGER = logging.getLogger(__name__)
-
-
-async def validate_input(hass: HomeAssistant, data: dict[str, str]) -> dict[str, str]:
-    """Validate the user input allows us to connect."""
+STEP_LOGIN_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
 
-    websession = async_get_clientsession(hass)
-    api = HabitipyAsync(
-        conf={
-            "login": data[CONF_API_USER],
-            "password": data[CONF_API_KEY],
-            "url": data[CONF_URL] or DEFAULT_URL,
-        }
-    )
-    try:
-        await api.user.get(session=websession)
-        return {
-            "title": f"{data.get('name', 'Default username')}",
-            CONF_API_USER: data[CONF_API_USER],
-        }
-    except ClientResponseError as ex:
-        raise InvalidAuth from ex
+_LOGGER = logging.getLogger(__name__)
 
 
 class HabiticaConfigFlow(ConfigFlow, domain=DOMAIN):
     """Handle a config flow for habitica."""
 
     VERSION = 1
 
-    async def async_step_user(self, user_input=None):
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
         """Handle the initial step."""
 
-        errors = {}
+        return self.async_show_menu(
+            step_id="user",
+            menu_options=["login", "advanced"],
```

Comment:
But both are technically logging in, so can we think of better names?

Reply:
habitica and self_hosted would do I think

Reply:
Legacy Login xD

Reply:
normies and the rest

Reply:
Maybe "Login to Habitica" and "Login to other instances" ??

Reply:
sounds good